msvc/mono.props

```diff
@@ -17,7 +17,8 @@
     <MONO_USE_STATIC_LIBMONO>false</MONO_USE_STATIC_LIBMONO>
   </PropertyGroup>
   <PropertyGroup Label="MonoDirectories">
-    <top_srcdir>$(MSBuildProjectDirectory)/..</top_srcdir>
+    <MonoSourceLocation Condition="'$(MonoSourceLocation)' == '' ">..</MonoSourceLocation>
+    <top_srcdir>$(MSBuildProjectDirectory)/$(MonoSourceLocation)</top_srcdir>
     <MONO_DIR>$(top_srcdir)</MONO_DIR>
```

Comment:
I don't believe that `top_srcdir` is needed for anything else, so indeed, we could eliminate `MonoSourceLocation` entirely and use `top_srcdir`. I think the name `MonoSourceLocation` is a bit more descriptive than `top_srcdir` when it is used in the .targets files though, so I would prefer to leave this as-is. I'm open to collapsing them into one variable though.